os/pm/pm_procfs.c

```diff
@@ -520,6 +527,92 @@ static size_t power_unlock_write(FAR struct file *filep, FAR const char *buffer,
 	return OK;
 }
 
+/****************************************************************************
+ * Name: power_set_next_wakeup_interval
+ *
+ * Description:
+ *  Sets timer interval for next sleep cycle after sending of wifi keep 
+ *  alive signal from UART. It also stores the current time tick 
+ *  in g_pm_timer.last_wakeup_time that will be used to calculate the 
+ *  duration of required sleep later just before sleep. If another
+ *  interrupt occurs during this interval, the system will still
+ *  wake up.
+ *  Also we dont want to do pm_set_timer to PM_WAKEUP_TIMER if the
+ *  system is already in PM_LOCK_TIMER state. If system is in 
+ *  PM_LOCK_TIMER state, then power_unlock_write can only change that state.
+ *
+ * Input Parameters:
+ *   filep          - File path of the power domain
+ *   buffer         - Not used
+ *   buflen         - Use of timer interrupt and its duration
+ *
+ * Returned Value:
+ *  0 (OK) means the state unlock was successful.
+ ****************************************************************************/
+ 
+static size_t power_set_next_wakeup_interval(FAR struct file *filep, FAR const char *buffer, size_t buflen)
+{
+	(void)buffer;
+
+	g_pm_timer.last_wakeup_time = clock_systimer();
+	if (g_pm_timer.timer_type == 0 && buflen > 0) {
```

Comment:
Since we have an enum for the time types
Use the enum name itself instead of 0 to directly indicate the timer type and make the code easier to understand,.

Reply:
Actually, 0 doesnot have any enum name. timer_type == 0 means timer has not set.

Reply:
0 then can be added as an enumerated value to avoid using the hardcoded value '0'?

Reply:
updated